include/lbann/base.hpp

```diff
@@ -44,6 +44,12 @@ namespace lbann {
 // Forward-declaration.
 class lbann_comm;
 
+// support for encoding data_id in conduit::Node, used by
+// conduit_data_store and associated code
+#define LBANN_SAMPLE_ID_PAD 9
```

Comment:
Do these need to be defined here? Can you move them to a conduit-specific header?

Reply:
they're used by data_store_conduit, and all data_readers that are (will be) supporting the data_store. I think they can be put in data_store_conduit.hpp